src/core_types.cpp

```diff
@@ -2,6 +2,24 @@
 
 namespace mls {
 
+hpke::Signature::ID
+hpke_signature_algorithm(SignatureScheme scheme)
```

Comment:
You should only need one direction of this conversion, `hpke -> tls`.  Just delete this method and in the check below:
```
BEFORE: to_hpke(cred._signature_scheme) != cipher_suite.get().sig.id
AFTER : cred._signature_scheme != to_tls(cipher_suite.get().sig.id)
```

I would also put that conversion in `crypto.h` / `crypto.cpp`.  That will expose `hpke::Signature::ID` in the short run, but (a) you should be able to forward-declare it, and (b) we can make it internal in the longer run.